File: data/Logger/Logger.java
import java.io.FileWriter;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.List;

public class Logger {
    private String logFile;

    public Logger(String logFile) {
        this.logFile = logFile;
    }

    // Write a log message to the file
    public void log(String message) {
        try (FileWriter writer = new FileWriter(logFile, true)) {
            writer.write(message + "\n");
            System.out.println("Logged: " + message);
        } catch (IOException e) {
            System.out.println("Error writing to log file: " + e.getMessage());
        }
    }

    // Read all logs from the file
    public void readLogs() {
        try {
            List<String> logs = Files.readAllLines(Paths.get(logFile));
            System.out.println("Logs:");
            logs.forEach(System.out::println);
        } catch (IOException e) {
            System.out.println("Error reading log file: " + e.getMessage());
        }
    }

    public static void main(String[] args) {
        Logger logger = new Logger("app.log");
        logger.log("Application started.");
        logger.log("User logged in.");
        logger.readLogs();
    }
}

File: data/Logger/Logger.cs
using System;
using System.IO;
using System.Collections.Generic;

public class Logger {
    private string logFile;

    public Logger(string logFile) {
        this.logFile = logFile;
    }

    // Write a log message to the file
    public void Log(string message) {
        try {
            using (StreamWriter writer = new StreamWriter(logFile, true)) {
                writer.WriteLine(message);
                Console.WriteLine("Logged: " + message);
            }
        } catch (Exception e) {
            Console.WriteLine("Error writing to log file: " + e.Message);
        }
    }

    // Read all logs from the file
    public void ReadLogs() {
        try {
            string[] logs = File.ReadAllLines(logFile);
            Console.WriteLine("Logs:");
            foreach (string log in logs) {
                Console.WriteLine(log);
            }
        } catch (Exception e) {
            Console.WriteLine("Error reading log file: " + e.Message);
        }
    }

    public static void Main(string[] args) {
        Logger logger = new Logger("app.log");
        logger.Log("Application started.");
        logger.Log("User logged in.");
        logger.ReadLogs();
    }
}